Src/FluentAssertions/Events/EventRecorder.cs

```diff
@@ -65,10 +65,10 @@ public void Attach(WeakReference subject, EventInfo eventInfo)
 
     public void Dispose()
     {
-        if (cleanup is not null)
+        if (this.cleanup is { } cleanup)
```

Comment:
I actually think this makes it _less_ readable.

Reply:
Would you prefer something like this?

```cs
Action localCleanup = cleanup;
if (localCleanup is not null)
{
    localCleanup();
    cleanup = null;
    EventObject = null;
    raisedEvents.Dispose();
}
```

Reply:
Yeah, even though that's ugly too, it's _less_ difficult to read than comparing with `{ }`